dataset/shape.cpp

```diff
@@ -141,4 +141,271 @@ Dataset resize(const DatasetConstView &d, const Dim dim,
   return result;
 }
 
+namespace {
+
+/// Split dims for reshaping one dim into multiple dims
```

Comment:
To me, `split` implies getting multiple outputs of the items passed to `split` i.e. https://numpy.org/doc/stable/reference/generated/numpy.split.html?highlight=split#numpy.split

I understand that `reshape` is too generic for this operation (at least as an understood term in numpy etc). 

Maybe something like `morph` would be better?

Reply:
`morph` is quite meaningless on its own, imho?

Reply:
yes we spent quite a while trying to find good names.
there was `split`, `fold`, `reshape`, but we thought that `split` worked the best in the end.

Reply:
I wouldn't mind `fold`? Would match nicely with `flatten` (you `fold` a sheet, and you later `flatten` it).

Reply:
Is part of the problem, that the names really need to account for the fact that you are applying to dimensions. i.e. `split_dims` might make more sense.

Reply:
Maybe just leave as it is. I don't have any strong suggestions for improving this.

Reply:
`fold` and `unfold`?

Reply:
I agree. Fold does have a good quality there when applied to dimensions.

Reply:
Not so sure. fold is often used as a synonym for reduce / accumulate. I think `split_dims` is more appropriate.

Reply:
But then it should maybe be `reshape_dim`, since it isn't really splitting into multiple outputs?

Reply:
Split is indeed a bit ambiguous, one might expect that you end up with 2 separate halves of a data array if you split a dimension.

Reply:
@jl-wynen does make a good point about reduction via `fold`. That is fairly common in other languages, but there is no name clash directly in python to my knowledge.

Reply:
I think we should consider:
1. How like is it that we would want to add a reduction named `fold`.
2. What fraction of our users would use this "reduction fold" compared to the "shape fold"

My uninformed feeling is that the "shape fold" is much more common, so I would vote for this name.

Reply:
Common on programming? I have never seen it in this context.
I don't think that calling it fold would introduce any real confusion. Python and numpy don't use it. And it is typically used in functional programming. Those people probably know what they are doing, so they should be able to cope.

Reply:
A couple more arguments against `split`:
- we write that `groupby` is doing a **split**-apply-combine operation, which is sifferent from splitting a dim.
- we have a function called `slices`, that returns a dict of slices along one dimension. maybe this should actually be called `split`?

Reply:
> Common on programming? I have never seen it in this context.

I think Simon meant that users of scipp will use the dimension reshape folding operation more than a potential fold reduction operation that we haven't defined yet, because the reshape is already needed in some of our notebooks